src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import org.folio.consortia.config.FolioExecutionContextHelper;
+import org.folio.consortia.domain.dto.PublicationRequest;
+import org.folio.consortia.domain.dto.SharingSettingRequest;
+import org.folio.consortia.domain.dto.SharingSettingResponse;
+import org.folio.consortia.domain.dto.Tenant;
+import org.folio.consortia.domain.dto.TenantCollection;
+import org.folio.consortia.repository.SharingSettingRepository;
+import org.folio.consortia.service.ConsortiumService;
+import org.folio.consortia.service.PublicationService;
+import org.folio.consortia.service.SharingSettingService;
+import org.folio.consortia.service.TenantService;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.scope.FolioExecutionContextSetter;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class SharingSettingServiceImpl implements SharingSettingService {
+
+  private final SharingSettingRepository sharingSettingRepository;
+  private final TenantService tenantService;
+  private final ConsortiumService consortiumService;
+  private final FolioExecutionContextHelper contextHelper;
+  private final PublicationService publicationService;
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  public SharingSettingResponse start(UUID consortiumId, SharingSettingRequest sharingSettingRequest) {
+    log.debug("start:: Trying to share setting with consortiumId: {}, sharing settingId: {}", consortiumId, sharingSettingRequest.getSettingId());
+    consortiumService.checkConsortiumExistsOrThrow(consortiumId);
+    Set<String> tenantsAssociationWithSetting = sharingSettingRepository.findTenantsBySettingId(sharingSettingRequest.getSettingId());
+    TenantCollection tenantCollection = tenantService.getAll(consortiumId);
+
+    PublicationRequest publicationRequestPostMethod = createPublicationRequestForSetting(sharingSettingRequest, "POST");
+    PublicationRequest publicationRequestPutMethod = createPublicationRequestForSetting(sharingSettingRequest, "PUT");
+
+    // By traverse through all tenants in db,
+    // we add tenant to put method publication tenant list, if it exists in setting tenant associations
+    // otherwise, we add it to post method publication tenant list
+    for (Tenant tenant : tenantCollection.getTenants()) {
+      if (tenantsAssociationWithSetting.contains(tenant.getId())) {
+        publicationRequestPutMethod.getTenants().add(tenant.getId());
+      } else {
+        publicationRequestPostMethod.getTenants().add(tenant.getId());
+      }
+    }
+    log.info("start:: tenants with size: {} successfully added to appropriate publication request", tenantCollection.getTotalRecords());
```

Comment:
please add log in which tenants you are going to create setting and in which - to update